src/react/components/common/cloudFilePicker/cloudFilePicker.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import { ICloudConnection, IConnection } from "../../../../models/applicationState";
+import { IStorageProvider, StorageProviderFactory } from "../../../../providers/storage/storageProvider";
+import CondensedList, { ListItem } from "../condensedList/condensedList";
+
+export interface ICloudFilePickerProps {
+    isOpen: boolean;
+    connections: IConnection[];
+    onCancel: () => void;
```

Comment:
Would you say `onCancel` is required?  If i don't care what happens should I still need to specify it?  If not let's make this an optional property.